learningequality/kolibri
# Quiz creation side panels improvements

## Summary


https://github.com/user-attachments/assets/47463a48-62f0-48d6-ac04-4d5455315b28


When working on quiz creation I created a SectionSidePanel component to kind of wrap around all of the various side-panel kinds but realized a bit too late on that it wasn't really necessary and added some unnecessary complexity.

This instead begins organizing the code so that each of the child routes' components for the side panels are wrapped in the SidePanelModal component which results in them being rendered into the `<router-view/>` in the quiz creation root route.

Some motivations for this were:

- Removing unnecessary indirection around when/how/where components define what to do re: back and close behavior
- Providing a better example for handling side panels w/ routing for Lessons
- Much easier to put things into the Side panel title section

## Reviewer guidance


### QA

- No regressions in the quiz creation side panels - section ordering, section editing, resource selection (search,bookmarks,etc), question replacement.
- Ensure that "are you sure" modals are triggered and that they're "Yes/No" handling works properly.
- Notably - "Section Editor <Click 'Add Questions'>  ->  Resource Selection" - make a change to your selection, then hit the browser back button until you get the "Are you sure" modal -- note that when you confirm it, you are returned correctly to the "Section Editor" side panel. However, had you clicked the "X" to close it, then you'd just go straight to the root page.


### Code review

- Whaddya think overall? 

----

## Testing checklist



## PR process


## Reviewer checklist

- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md
